stan/math/prim/err/is_infinity.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef STAN_MATH_PRIM_ERR_IS_INFINITY_HPP
+#define STAN_MATH_PRIM_ERR_IS_INFINITY_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/fun/constants.hpp>
+#include <stan/math/prim/fun/get.hpp>
+#include <stan/math/prim/fun/size.hpp>
+#include <stan/math/prim/fun/value_of_rec.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return <code>true</code> if <code>y</code> is infinite.
+ * This function is vectorized and will check each element of
+ * <code>y</code>.
+ * @tparam T_y Type of y
+ * @param y Variable to check
+ */
+template <typename T_y, require_matrix_t<T_y>* = nullptr>
+inline bool is_infinity(const T_y& y) {
+  return (y.array() == INFTY).any();
```

Comment:
Call it `is_positive_infinity` (and update docs to reflect this) since usually infinity checks include negative and positive.